src/panels/lovelace/header-footer/hui-graph-header-footer.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  html,
+  LitElement,
+  TemplateResult,
+  customElement,
+  property,
+  PropertyValues,
+} from "lit-element";
+
+import "../components/hui-graph-base";
+
+import { LovelaceHeaderFooter } from "../types";
+import { HomeAssistant } from "../../../types";
+import { GraphHeaderFooterConfig } from "./types";
+import { getHistoryCoordinates } from "../common/graph/get-history-coordinates";
+
+const minute = 60000;
+
+@customElement("hui-graph-header-footer")
+export class HuiGraphHeaderFooter extends LitElement
+  implements LovelaceHeaderFooter {
+  public static getStubConfig(): object {
+    return {};
+  }
+
+  @property() public hass?: HomeAssistant;
+  @property() protected _config?: GraphHeaderFooterConfig;
+  @property() private _coordinates?: any;
+  private _date?: Date;
+
+  public setConfig(config: GraphHeaderFooterConfig): void {
+    if (!config?.entity || config.entity.split(".")[0] !== "sensor") {
+      throw new Error(
+        "Invalid Configuration: An entity from within the sensor domain required"
+      );
+    }
+
+    const cardConfig = {
+      detail: 1,
+      hours_to_show: 24,
+      ...config,
+    };
+
+    cardConfig.hours_to_show = Number(cardConfig.hours_to_show);
+    cardConfig.detail =
+      cardConfig.detail === 1 || cardConfig.detail === 2
+        ? cardConfig.detail
+        : 1;
+
+    this._config = cardConfig;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+
+    const stateObj = this.hass!.states[this._config.entity];
+
+    if (!stateObj.attributes.unit_of_measurement) {
```

Comment:
Why do we need a `unit_of_measurement` for a line graph?

Reply:
No 100% sure. That's how we built it back when we built the sensor card. I tried a sensor with no UOM and received an error. but the state was also not a number.

Added in a catch for if the state history is undefined. That should catch this.